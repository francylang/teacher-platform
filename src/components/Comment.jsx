import React, { Component } from 'react';

class Comment extends Component {
  render() {
    if (this.props.commentId === this.props.discussionId) {
      return (
        <article className="comment-section">
          <h4 className="comments-header">Comments:</h4>
          <p className="comment">{this.props.comment}</p>
          <ul>
            <li>From: Amy</li>
            <li>Date: December 28, 2018</li>
          </ul>
        </article>
      )
    } else {
      return null;
    }
  }
}

export default Comment;
