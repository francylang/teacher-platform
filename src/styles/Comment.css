.comments-header {
  text-align: left;
  margin-bottom: 0;
  font-size: 100%; }

.comment {
  text-align: left;
  padding-bottom: 8px;
  border-bottom: 1px solid #A2A9AD; }

ul {
  list-style: none;
  text-align: left;
  padding-left: 0; }

li {
  color: #5D5B5E; }
