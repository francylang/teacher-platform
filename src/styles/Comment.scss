.comment-section {
  margin-top: 50px;
}

.comments-header {
  text-align: left;
}

.comment {
  text-align: left;
  padding-top: 8px;
  border-top: 1px solid #6c6ae5;
}

ul {
  list-style: none;
  text-align: left;
  padding-left: 0;
}

li {
    color: #5D5B5E;
}
