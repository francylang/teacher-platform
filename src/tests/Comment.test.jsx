import React from 'react';
import ReactDOM from 'react-dom';
import Comment from '../components/Comment';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<Comment />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Comment />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a comment section', () => {
    const comment = wrapper.find('.comment-section');

    expect(comment.length).toEqual(1);
  });

  it('should render a comment header', () => {
    const header = wrapper.find('.comments-header');

    expect(header.length).toEqual(1);
  });

  it('should render comments', () => {
    const comment = wrapper.find('.comment');

    expect(comment.length).toBeDefined();
  });

  it('should render a list with comment info', () => {
    const list = wrapper.find('ul');

    expect(list.length).toBeDefined();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
