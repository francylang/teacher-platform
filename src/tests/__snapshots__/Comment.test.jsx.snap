// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Comment />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4
          className="comments-header"
>
          Comments:
</h4>,
        <p
          className="comment"
>
          I am the comment and I am a really great comment!!!
</p>,
        <ul>
          <li>
                    From: Amy
          </li>
          <li>
                    Date: December 28, 2018
          </li>
</ul>,
        <p
          className="comment"
>
          I am ANOTHER comment and I am a really great comment!!!
</p>,
        <ul>
          <li>
                    From: Amy
          </li>
          <li>
                    Date: December 28, 2018
          </li>
</ul>,
      ],
      "className": "comment-section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Comments:",
          "className": "comments-header",
        },
        "ref": null,
        "rendered": "Comments:",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am the comment and I am a really great comment!!!",
          "className": "comment",
        },
        "ref": null,
        "rendered": "I am the comment and I am a really great comment!!!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              From: Amy
</li>,
            <li>
              Date: December 28, 2018
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From: Amy",
            },
            "ref": null,
            "rendered": "From: Amy",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Date: December 28, 2018",
            },
            "ref": null,
            "rendered": "Date: December 28, 2018",
            "type": "li",
          },
        ],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "I am ANOTHER comment and I am a really great comment!!!",
          "className": "comment",
        },
        "ref": null,
        "rendered": "I am ANOTHER comment and I am a really great comment!!!",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              From: Amy
</li>,
            <li>
              Date: December 28, 2018
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "From: Amy",
            },
            "ref": null,
            "rendered": "From: Amy",
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Date: December 28, 2018",
            },
            "ref": null,
            "rendered": "Date: December 28, 2018",
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4
            className="comments-header"
>
            Comments:
</h4>,
          <p
            className="comment"
>
            I am the comment and I am a really great comment!!!
</p>,
          <ul>
            <li>
                        From: Amy
            </li>
            <li>
                        Date: December 28, 2018
            </li>
</ul>,
          <p
            className="comment"
>
            I am ANOTHER comment and I am a really great comment!!!
</p>,
          <ul>
            <li>
                        From: Amy
            </li>
            <li>
                        Date: December 28, 2018
            </li>
</ul>,
        ],
        "className": "comment-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Comments:",
            "className": "comments-header",
          },
          "ref": null,
          "rendered": "Comments:",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I am the comment and I am a really great comment!!!",
            "className": "comment",
          },
          "ref": null,
          "rendered": "I am the comment and I am a really great comment!!!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                From: Amy
</li>,
              <li>
                Date: December 28, 2018
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From: Amy",
              },
              "ref": null,
              "rendered": "From: Amy",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date: December 28, 2018",
              },
              "ref": null,
              "rendered": "Date: December 28, 2018",
              "type": "li",
            },
          ],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "I am ANOTHER comment and I am a really great comment!!!",
            "className": "comment",
          },
          "ref": null,
          "rendered": "I am ANOTHER comment and I am a really great comment!!!",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li>
                From: Amy
</li>,
              <li>
                Date: December 28, 2018
</li>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "From: Amy",
              },
              "ref": null,
              "rendered": "From: Amy",
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Date: December 28, 2018",
              },
              "ref": null,
              "rendered": "Date: December 28, 2018",
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
